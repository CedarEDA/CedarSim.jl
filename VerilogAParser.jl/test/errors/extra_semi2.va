`define MAX_EXP 5.834617425e14
`define MIN_EXP 1.713908431e-15
`define EXP_THRESHOLD 34.0

module extra_semi2(p, n);
    analog function real expclamp;
        input x;

        begin
            if (x > `EXP_THRESHOLD)
                expclamp = `MAX_EXP;
            else if (x < `EXP_THRESHOLD)
                expclamp = `MIN_EXP;
            else
                expclamp = exp(x);
        end;
    endfunction
endmodule