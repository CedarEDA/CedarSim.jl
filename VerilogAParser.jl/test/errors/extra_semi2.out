[31m[1mâ•­ ERROR [VAS46]:[22m[39m[31m Unexpected semicolon.[39m
[31mâ”‚[39m[90m   â•­â”€â”€â”€â”€â”€ extra_semi2.va:9[39m
[31mâ”‚[90m9[0m  [90mâ”‚[0m         begin
[31mâ”‚[90m10[0m [90mâ”‚[0m             if (x > `EXP_THRESHOLD)
[31mâ”‚[90m11[0m [90mâ”‚[0m                 expclamp = `MAX_EXP;
[31mâ”‚[90m12[0m [90mâ”‚[0m             else if (x < `EXP_THRESHOLD)
[31mâ”‚[90m13[0m [90mâ”‚[0m                 expclamp = `MIN_EXP;
[31mâ”‚[90m14[0m [90mâ”‚[0m             else
[31mâ”‚[90m15[0m [90mâ”‚[0m                 expclamp = exp(x);
[31mâ”‚[90m16[0m [90mâ”‚[0m         end[91m;
[31mâ”‚[0m   [90mâ”Š[0m            [91mâ”“
[31mâ”‚[0m   [90mâ”Š[0m [91mâœ— Unexpected semicolon[0m
[31mâ”‚[39m[90m   â”‚[39m
[31mâ”‚[39m[90m   â•°â”€â”€â”€â”€â”€ [39mUnexpected semicolon.
